### PR TITLE
Refactor Localization to *not* reference Bundle.main and Bundle(for:)

### PR DESCRIPTION
**Pull request by syoung-smallwisdom**
@kalperin 

This is the last (I think) of the changes to support fully separating the platform context into a separate "module" as a precursor to using Swift Package Manager.

### CODE REVIEW

**Comment by kalperin** [comment]
> @kalperin
> 
> This is the last (I think) of the changes to support fully separating the platform context into a separate "module" as a precursor to using Swift Package Manager.



@Linziil @Shalini-Patel

File: Research/Research.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1078,6 +1078,10 @@
 		FF8B54EE1FCE6D15006B6937 /* RSDExceptionHandler.m in Sources */ = {isa = PBXBuildFile; fileRef = FFF159DF1FB4D7A60061BA93 /* RSDExceptionHandler.m */; };
 		FF9557CC23060F4B00B8D60B /* RSDAppDelegate.swift in Sources */ = {isa = PBXBuildFile; fileRef = FF9557CB23060F4B00B8D60B /* RSDAppDelegate.swift */; };
 		FF9C264B243FA48B00241202 /* QuestionTableDataSourceTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = FF9C264A243FA48B00241202 /* QuestionTableDataSourceTests.swift */; };
+		FF9F343724EB473600736F04 /* LocalizationBundle.swift in Sources */ = {isa = PBXBuildFile; fileRef = FF9F343624EB473600736F04 /* LocalizationBundle.swift */; };
```
Comment (Erin-Mounts) [inline Research/Research.xcodeproj/project.pbxproj:1081]: looks like somehow this got added 4x to the project file...

File: Research/Research/Localization/Localization.swift [hunk 2]
```diff
@@ -136,6 +63,18 @@ open class Localization: NSObject {
             allBundles.append(bundle)
         }
     }
+    
+    /// Insert a bundle into `allBundles` at a given index. If the index is beyond the range of
+    /// `allBundles`, then the bundle will be appended to the end of the array. If the bundle was
+    /// previously in the array, then the previous instance will be moved to the new .
```
Comment (Erin-Mounts) [inline Research/Research/Localization/Localization.swift]: to the new what

File: Research/Research/PlatformContext/Localization/LocalizationBundle.swift [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    
+    /// Convenience initializer that uses the default `tableName` and `targetSuffixes`.
+    /// - parameter bundle: The bundle to check for a localized string.
+    public convenience init(_ bundle: Bundle) {
+        let tableName = bundle.bundleIdentifier?.components(separatedBy: ".").last
+        let targetSuffixes = ["-iOS", "-tvOS", "-watchOS", "-macOS"]
+        self.init(bundle: bundle, tableName: tableName, targetSuffixes: targetSuffixes)
+    }
+    
+    /// Convenience initializer that uses the default `tableName` and `targetSuffixes`.
```
Comment (Erin-Mounts) [inline Research/Research/PlatformContext/Localization/LocalizationBundle.swift]: this one doesn't use the defaults though

**Review by Erin-Mounts — APPROVED** [review]
couple of comments/nits, otherwise 👍

File: Research/Research.xcodeproj/project.pbxproj (hunk 1 above)
Comment (syoung-smallwisdom) [reply Research/Research.xcodeproj/project.pbxproj:1081]: Yes -> macOS, tvOS, iOS, watchOS

**Pull request merged**